dwh/indicator_lib.py

```diff
@@ -82,14 +82,48 @@ def calc_TX_PVLS(patient_agg_obs: pandas.DataFrame, VL_code: str,
   if end_date_str:
     end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
   temp_df = patient_agg_obs[(patient_agg_obs['code'] == VL_code)].copy()
+  # separate value from date
+  temp_df[['vl_max_date', 'latest_vl_value']] = temp_df.max_date_value.str.split('_SeP_', expand=True)
+  temp_df['latest_vl_value'] = temp_df['latest_vl_value'].astype(float)
   # Note the above copy is used to avoid setting a new column on a slice next:
-  temp_df['sup_VL'] = (temp_df['max_value'] < failure_threshold)
+  temp_df['sup_VL'] = (temp_df['latest_vl_value'] < failure_threshold)
   temp_df['buckets'] = temp_df.apply(
       lambda x: _agg_buckets(x.birthDate, x.gender, datetime.today()), axis=1)
   temp_df_exp = temp_df.explode('buckets')
   temp_df_exp = temp_df_exp.groupby(['sup_VL', 'buckets'], as_index=False)\
       .count()[['sup_VL', 'buckets', 'patientId']]\
-      .rename(columns={'patientId': 'count'})
+      .rename(columns={'patientId': 'sup_VL_count'})
   # calculate ratio
-  temp_df_exp['ratio'] = temp_df_exp['count']/temp_df['sup_VL'].count()
+  temp_df_exp['sup_VL_ratio'] = temp_df_exp['sup_VL_count']/temp_df['sup_VL'].count()
   return temp_df_exp
+
+
+def calc_TX_NEW(patient_agg_obs:  pandas.DataFrame, ARV_plan: str, start_drug: List,
+    end_date_str: str = None) -> pandas.DataFrame:
+  """Calculates TX_NEW indicator with its corresponding disaggregations.
+  TX_NEW indicator counts the number of adults and children newly enrolled
+  on antiretroviral therapy (ART) within the provided
+  Args:
+    patient_agg_obs: A DataFrame generated by `patient_query.find_patient_aggregates()`.
+    ARV_plan: The concept question code for ANTIRETROVIRAL PLAN
+    start_drug: The concept answer code for START DRUG
+    end_date: The string representation of the last date as 'YYYY-MM-DD'.
+  Returns:
+  """
+  end_date = datetime.today()
+  if end_date_str:
+    end_date = datetime.strptime(end_date_str, '%Y-%m-%d')
+  temp_df = patient_agg_obs[(patient_agg_obs['code'] == ARV_plan)].copy()
+  # separate value from date
+  temp_df[['ARV_max_date', 'latest_ARV_plan']] = temp_df.max_date_value_code.str.split('_SeP_', expand=True)
+  # Note the above copy is used to avoid setting a new column on a slice next:
+  temp_df['TX_NEW'] = (temp_df['latest_ARV_plan'].isin(start_drug))
+  temp_df['buckets'] = temp_df.apply(
```

Comment:
Please refactor the aggregation section (i.e., line 121 to 128) from this and the previous functions.

Reply:
Any thoughts about this request?